### PR TITLE
Bump `cookiecutter` to address known vulnerability

### LINKED ISSUE #1600: Security issue with cookiecutter<2.1.1
## Description
Snyk is complaining about a security issue with the ``cookiecutter``  (1.7.0) used in kedro. I don't know if it is valid or not, but it breaks my CI workflow when installing kedro. 

image
 

## Context

``kedro`` cannot be installed in an enterprise context, and my [kedro-mlflow CI](https://github.com/Galileo-Galilei/kedro-mlflow/pull/327) complains. This impacts all kedro versions. 

## Other remarks

I haven't seen a specific process to warn about potential security issues, but it does not feel right to open a public github issue though, did I miss something there? This one does not seem too big a threat (and snyk warns publicly about it), so I choose to report it there.

### PR DESCRIPTION
**Pull request by deepyaman**
Signed-off-by: Deepyaman Datta <deepyaman.datta@utexas.edu>

## Description
Closes https://github.com/kedro-org/kedro/issues/1414 and closes #1600.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1601"></a>

### CODE REVIEW

**Comment by AntonyMilneQB** [comment]
Looking at the [cookiecutter releases](https://github.com/cookiecutter/cookiecutter/releases), it seems that 2.0.x never made it to PyPI. Releases went 1.7.3, 2.1.0, 2.1.1.

Looking at the breaking changes in 2.1.0 I don't think any should affect us, and all the tests pass here, but given how central cookiecutter is to kedro I think we should do a quick manual test to make sure that various `kedro new` commands are still working.

File: requirements.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
 anyconfig~=0.10.0
 cachetools~=4.1
 click<9.0
-cookiecutter~=1.7.0
-dynaconf>=3.1.2,<4.0.0
+cookiecutter>=2.1.1
```
Comment (AntonyMilneQB) [inline requirements.txt:4]: Should we add an upper bound of `<3`?

Comment (AntonyMilneQB) [reply requirements.txt:4]: Yeah, after skimming through [this article](https://iscinumpy.dev/post/bound-version-constraints/), similar things elsewhere and from some recent experiences I think we should revisit the way we pin versions also. The bounds on CalVer packages we have should make sense though: black actually has [exactly the stability policy](https://black.readthedocs.io/en/stable/the_black_code_style/index.html#stability-policy) that you mention (although it doesn't really do "breaking" changes the same way at all). Other packages that are CalVer we have a kind of lagging pin where we upper bound to the last version we saw it was working.

So for now I would add `<3` just to be consistent with how we do other requirements, even though I think we should reconsider that general policy.

Comment (deepyaman) [reply requirements.txt:4]: > black actually has [exactly the stability policy](https://black.readthedocs.io/en/stable/the_black_code_style/index.html#stability-policy) that you mention

TIL!

File: requirements.txt [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
+dynaconf>=3.1.2, <4.0.0
```
Comment (deepyaman) [inline requirements.txt:5]:
```suggestion
dynaconf>=3.1.2, <4.0
```

File: requirements.txt (hunk 1 above)
Comment (deepyaman) [inline requirements.txt:4]:
```suggestion
cookiecutter>=2.1.1, <3.0
```

**Review by AntonyMilneQB — APPROVED** [review]
Thank you for this! Happy to approve assuming that you manually tested it and it works ok 😀

**Comment by deepyaman** [comment]
> Looking at the [cookiecutter releases](https://github.com/cookiecutter/cookiecutter/releases), it seems that 2.0.x never made it to PyPI. Releases went 1.7.3, 2.1.0, 2.1.1.
> 
> Looking at the breaking changes in 2.1.0 I don't think any should affect us, and all the tests pass here, but given how central cookiecutter is to kedro I think we should do a quick manual test to make sure that various `kedro new` commands are still working.

I did `kedro new` and `kedro new --starter spaceflights` with both the current and latest `cookiecutter` versions, and the output is identical.

**Pull request merged**